### DISCUSSION TITLE
Codemagic API Returning Deleted GitHub Branches After Deletion

### DISCUSSION THREAD

**Discussion by Imgkl** [codemagic-ci-cd/codemagic-docs#2762] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. **Set up a repository**: 
   - Ensure that you have a GitHub repository connected to Codemagic, with multiple branches.

2. **Trigger API call**:
   - Make a call to the Codemagic API endpoint `https://api.codemagic.io/apps` to retrieve the list of branches for the repository.
   - Verify that all the branches in the repository are reflected in the API response.

3. **Delete a branch on GitHub**:
   - Go to the GitHub repository and delete one or more branches.

4. **Check API response again**:
   - Without refreshing the Codemagic dashboard, make another call to the Codemagic API to retrieve the list of branches.
   - Observe that the deleted branches are still being returned in the API response.

5. **Refresh Codemagic dashboard**:
   - Manually refresh the Codemagic dashboard by going to the app's page.
   - After the dashboard refresh, make another API call to fetch the branches.

6. **Compare results**:
   - Notice that after refreshing the dashboard, the deleted branches are no longer present in the API response, indicating the branch list is only updated after a manual refresh on the dashboard.

### Expected results

- When a branch is deleted from the connected GitHub repository, the Codemagic API should reflect this change immediately.
- The deleted branches should no longer appear in the API response after deletion, without requiring a manual refresh of the Codemagic dashboard.

### Actual results

- After deleting a branch from the connected GitHub repository, the Codemagic API continues to return the deleted branch in the API response.
- The deleted branch remains visible in the API response until a manual refresh of the Codemagic dashboard is performed.

**Comment by dtrdic** [comment]
Try by refreshing the repository connection, click on `Update your repository URL` button in the repository settings, then check again if it's returning the deleted branches.
![Screenshot 2024-10-07 at 10 29 03](https://github.com/user-attachments/assets/7a0cba62-47d1-4ccf-93c1-5100657e9f4c)

**Comment by Imgkl** [comment, marked as answer]
Hey @dtrdic,

Thanks for pointing this out. Though I was in need of API solution where we refresh and get the updated git branches through API. I was able to get the endpoint through the above mentioned. 


For anyone who is looking for a solution for the same issue. Here is the cURL:

```curl
curl --location --request POST 'https://api.codemagic.io/apps/${APP_ID}/branches' \
--header 'Content-Type: application/json' \
--header 'x-auth-token: <API-TOKEN>'
````

Replace the `APP_ID` and `API_TOKEN` with the appropriate values, then you are good to go.
